access_control.py

```diff
@@ -0,0 +1,225 @@
+
+Main logic lives in conda-smithy. This is just a wrapper for admin-requests infra.
+"""
+import os
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+
+import requests
+from ruamel.yaml import YAML
+
+GH_ORG = os.environ.get("GH_ORG", "conda-forge")
+
+CIRUN_FILENAME_RESOURCE_MAPPING = {
+    "cirun-gpu-runner": {
+        "resource": "cirun-gpu-runner",
+    },
+    "cirun-gpu-runner-pr": {
+        "resource": "cirun-gpu-runner",
+        "policy_args": ["pull_request"]
+    }
+}
+
+
+ACCESS_YAML_FILENAME = ".access_control.yml"
+
+
+def _get_access_control_files(path):
+    access_path = Path(path)
+    all_files = list(access_path.glob("*"))
+    return (
+        [f for f in all_files if f.parts[-1] != "example.txt"]
+    )
+
+
+def _get_filename_feedstock_mapping(path):
+    access_files = _get_access_control_files(path)
+    file_name_feedstock_mapping = {}
+    for file_path in access_files:
+        with open(file_path, "r") as f:
+            feedstocks = f.readlines()
+            feedstocks = [feedstock.strip() for feedstock in feedstocks]
+            file_name = file_path.parts[-1].strip(".txt")
+            file_name_feedstock_mapping[file_name] = feedstocks
+    return file_name_feedstock_mapping
+
+
+def _process_access_control_requests(path, remove=True):
+    print(f"Processing access control requests for {path}")
+    grant_access_request = _get_filename_feedstock_mapping(path)
+    for filename, feedstocks in grant_access_request.items():
+        if filename in CIRUN_FILENAME_RESOURCE_MAPPING:
+            resource_mapping = CIRUN_FILENAME_RESOURCE_MAPPING.get(filename)
+            resource = resource_mapping.get("resource")
+            policy_args = resource_mapping.get("policy_args")
+            for feedstock in feedstocks:
+                feedstock_repo = f"{feedstock}-feedstock"
+                print(f"Processing feedstock for access control: {feedstock_repo}")
+                _process_request_for_feedstock(
+                    feedstock_repo, resource, remove=remove, policy_args=policy_args
+                )
+                action = "remove" if remove else "add"
+                update_access_yaml(resource, feedstock_repo, action=action)
+
+
+def process_access_control_requests():
+    """Process access control requests"""
+    print("Processing access control request")
+    _process_access_control_requests("grant_access", remove=False)
+    _process_access_control_requests("revoke_access")
+
+
+def _process_request_for_feedstock(feedstock, resource, remove, policy_args):
+    feedstock_clone_path = tempfile.mkdtemp()
+    assert GH_ORG
+    clone_cmd = f"git clone --depth 1 https://github.com/{GH_ORG}/{feedstock}.git {feedstock_clone_path}"
+    print(f"Cloning: {clone_cmd}")
+    subprocess.run(
+        clone_cmd,
+        shell=True,
+    )
+
+    register_ci_cmd = [
+        "conda-smithy register-ci",
+        f"--organization {GH_ORG}",
```

Comment:
```suggestion
        f"--organization={GH_ORG}",
```
Or separate it in two tokens. But this will fail if separated by a space because it will be taken as a single command.

Reply:
Ah no wait, you are joining everything after. Hm, in that case why create a list to begin with?

Reply:
Only for readability purpose